anchore_engine/common/helpers.py

```diff
@@ -29,15 +29,23 @@ def make_response_error(errmsg, in_httpcode=None, details=None):
 
     if isinstance(errmsg, Exception):
         if 'anchore_error_json' in errmsg.__dict__:
-            if {'message', 'httpcode', 'detail'}.issubset(set(errmsg.__dict__['anchore_error_json'])):
-                ret.update(errmsg.__dict__['anchore_error_json'])
+            # Try to load it as json
+            try:
+                err_json = json.loads(errmsg.__dict__['anchore_error_json'])
+            except:
```

Comment:
I know that bare `try/except` blocks are pervasive around the Anchore Engine codebase, but we should be very careful on where/why we are doing these. At the very least we should log the exception if we truly don't have any idea what exception is going to be raised.

In this particular case a `json.loads()` is being called along with `errmsg`. For the JSON part, the exception to catch would be `ValueError`, the `dict` could either have a `.get()` or catch `KeyError` as well

Reply:
Addressed!